src/main/java/by/onliner/page/ShoppingCartPage.java

```diff
@@ -54,11 +73,72 @@ public String getRemovedItemInformation() {
     /***
      * Get a message from empty cart.
      */
-    @Step("Read the empty cart message")
+    @Step("Get the empty cart message")
     public String getEmptyCartMassage() {
         DriverManager.getDriver().navigate().refresh();
         Waiter.waitForVisibility(emptyCartMessage);
         logger.info(String.format("Displayed [%s] message", emptyCartMessage.getText()));
         return emptyCartMessage.getText();
     }
+
+    /***
+     * Get the item name from shopping cart list.
+     * @return name of the item
+     */
+    @Step("Get the item name")
+    public String getItemNameText() {
+        Waiter.waitForVisibility(itemNameText);
+        logger.info(String.format("Read the item name [%s] from shopping cart", itemNameText.getText()));
+        return itemNameText.getText().trim();
+    }
+
+    /***
+     * Check the complete order button fo visibility.
+     * @return visibility of button
+     */
+    @Step("Chek [Перейти к оформлению] button for visibility")
+    public boolean isCompleteOrderButtonVisible() {
+        boolean visibility = completeOrderButton.isDisplayed();
+        logger.info(String.format("'Перейти к оформлению' button visibility is {%s}", visibility));
+        return visibility;
+    }
+
+    /***
+     * Add one more same item in the shopping cart by [+] button
+     */
+    @Step("Add one more same item in the shopping cart by [+] button")
+    public void clickQuantityInputPlusButton() {
+        Waiter.elementToBeClickable(quantityInputPlusButton);
+        quantityInputPlusButton.click();
+        logger.info("Click '+' button");
+        Waiter.sleep(2);
+    }
+
+    /***
+     * Get number from quantity input
+     * @return integer number from quantity input
+     */
+    @Step("Get number from quantity input")
+    public int getNumberFromQuantityInput() {
+        Waiter.waitForVisibility(quantityInput);
+        logger.info("Get number [" + quantityInput.getAttribute("value") + "] from quantity input");
+        return Integer.parseInt(quantityInput.getAttribute("value"));
+    }
+
+    /***
+     * Get price of item
+     * @return integer price of item
+     */
+    @Step("Get price")
+    public int getPrice() {
+        Waiter.waitForVisibility(itemPriceText);
+        Pattern pattern = Pattern.compile("^[0-9]+");
+        Matcher matcher = pattern.matcher(itemPriceText.getText());
+        int price = 0;
+        while (matcher.find()) {
+            price = Integer.parseInt(matcher.group());
```

Comment:
is there any way to rewrite that without reinitializing price variable inside while?
Can we return the price value inside while? you can remove below logging if that's the only blocker